### DISCUSSION TITLE
'y.to' is missing in props validation

### DISCUSSION THREAD

**Discussion by esau-morais** [pmndrs/react-spring#2053] [Support]
I'm using [react-spring][1] to create a stacked cards component animation on scroll and basically it seems like it is applying the expected styling, but I'm getting this error down below. 

[![App.tsx file focusing on line 52, which has an error 'y.to' is missing in props validation][2]][2]

  [1]: https://www.react-spring.dev
  [2]: https://i.stack.imgur.com/1WbaT.jpg

The code basically pass through reference a list of card that has `imageUrl` and `id` as its props so when user uses the scroll, it gets to manipulate the `transform` CSS property of the elements by its position

```ts
import { useEffect, useRef } from 'react'

import { useSpring, animated } from '@react-spring/web'

const cards = [
  {
    id: 1,
    imageUrl: 'https://codyhouse.co/app/assets/img/stacking-cards-img-1.jpg'
  },
  {
    id: 2,
    imageUrl: 'https://codyhouse.co/app/assets/img/stacking-cards-img-2.jpg'
  },
  {
    id: 3,
    imageUrl: 'https://codyhouse.co/app/assets/img/stacking-cards-img-3.jpg'
  },
]

const StackedCards = () => {
  const cardsRef = useRef<HTMLDivElement[]>([])
  const [props, set] = useSpring(() => ({
    from: { y: 0 },
    to: { y: 0 }
  }))

  useEffect(() => {
    const observer = new IntersectionObserver((entries) => {
      entries.forEach((entry, index) => {
        if (entry.isIntersecting) {
          set({ y: index + 1 })
        } else {
          set({ y: 0 })
        }
      })
    })

    cardsRef.current.forEach((card) => {
      observer.observe(card)
    })
  }, [])


  return (
    <ul className='my-16 w-full flex flex-col gap-4'>
      {cards.map((card) => (
        <animated.div
          key={card.id}
          ref={(el) => cardsRef.current[card.id] = el as HTMLDivElement}
          className='transition-transform w-full top-5 overflow-hidden rounded-lg sticky origin-[center_top] mb-lg'
          style={{
            transform: props.y.to((y) => `translateY(-${10 * y}px)`),
          }}
        >
          <img className='aspect-[2/1] abolute top-0 left-0 w-full h-full object-cover' src={card.imageUrl} alt="" />
        </animated.div>
      ))
      }
    </ul >
  )
}

export default StackedCards
```

**Comment by joshuaellis** [comment]
Can you share a codesandbox demonstrating your issue?

**Reply by esau-morais** [reply]
Here's [the link](https://codesandbox.io/p/github/esau-morais/stackedCards.ts/draft/clever-kapitsa?file=%2Fsrc%2Fcomponents%2Fmolecules%2FStackedCards.tsx&workspace=%257B%2522activeFileId%2522%253A%2522clcyqb1sv000l8rh5b18w9cs2%2522%252C%2522openFiles%2522%253A%255B%2522%252FREADME.md%2522%252C%2522%252Fsrc%252Fcomponents%252Fmolecules%252FStackedCards.tsx%2522%255D%252C%2522sidebarPanel%2522%253A%2522GIT%2522%252C%2522gitSidebarPanel%2522%253A%2522COMMIT%2522%252C%2522spaces%2522%253A%257B%2522clcyqb1e600133i6j7iw72z9o%2522%253A%257B%2522key%2522%253A%2522clcyqb1e600133i6j7iw72z9o%2522%252C%2522name%2522%253A%2522Default%2522%252C%2522devtools%2522%253A%255B%257B%2522type%2522%253A%2522PREVIEW%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A3000%252C%2522key%2522%253A%2522clcyqb9v200bc3i6jvcqpti0z%2522%252C%2522isMinimized%2522%253Afalse%257D%252C%257B%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522key%2522%253A%2522clcyqb8zy007j3i6j6c1h203d%2522%252C%2522isMinimized%2522%253Atrue%257D%255D%257D%257D%252C%2522currentSpace%2522%253A%2522clcyqb1e600133i6j7iw72z9o%2522%252C%2522spacesOrder%2522%253A%255B%2522clcyqb1e600133i6j7iw72z9o%2522%255D%257D)

**Reply by esau-morais** [reply]
It seems more like an ESLint error related to prop-types. Once I disabled it, the error stopped being displayed, even tho I don't know the real problem and deslike the way I solved that.
![turned off react-/prop-types in .eslintrc.json](https://user-images.githubusercontent.com/55207584/212735785-22596da6-9b9a-4a14-a523-cb2151890446.png)

**Reply by joshuaellis** [reply]
Can't access that link because it's a project – i would create a minimal reproduction in a sandbox and share that :)
